### PR TITLE
adding the testing files and doing some modifications to the existing…

### PR DESCRIPTION
**Pull request by SallyElHajjar**
I have added the testing files that correspond to the previously added functions.
I have also modified the file data_utils to deal with tensorflow.
While I am not working with the data pipeline  "usl_lib CI" of the check shows one error when running "Run pytest usl_pipeline/usl_lib".  @Katsutoshii do you hjave any idea about why this is happening?

### CODE REVIEW

**Comment by Katsutoshii** [comment]
> I have added the testing files that correspond to the previously added functions. I have also modified the file data_utils to deal with tensorflow. While I am not working with the data pipeline "usl_lib CI" of the check shows one error when running "Run pytest usl_pipeline/usl_lib". @Katsutoshii do you hjave any idea about why this is happening?

I reran the checks and they seem to be passing now. Perhaps it was just some transient error.

File: usl_models/usl_models/atmo_ml/data_utils.py [hunk 1]
```diff
@@ -51,10 +54,19 @@ def split_time_step_pairs(data: tf.Tensor) -> tf.Tensor:
...
     """
-    # We need to permute the tensor to place the time and channel axes together
-    # during the reshape, in order to avoid disturbing the spatial dimensions.
+    # Transpose to [B, H, W, T, 2C] so that we can split the channel dimension
     permuted = tf.transpose(data, (0, 2, 3, 1, 4))
-    shape = permuted.shape[:3] + (permuted.shape[-2] * 2, permuted.shape[-1] // 2)
-    reshaped = tf.reshape(permuted, shape)
+
+    # Get the dynamic shape
+    batch_size = tf.shape(permuted)[0]
```
Comment (Katsutoshii) [inline usl_models/usl_models/atmo_ml/data_utils.py:61]: I generally prefer unpacking these in one line, like:
```
B, H, W, T_HALF, C_PAIR= tf.shape(permuted)
timesteps = 2 * T
channels = C // 2
```

**Review by Katsutoshii — APPROVED** [review]

File: usl_models/usl_models/atmo_ml/data_utils.py (hunk 1 above)
Comment (SallyElHajjar) [reply usl_models/usl_models/atmo_ml/data_utils.py]: Done!

Comment (SallyElHajjar) [reply usl_models/usl_models/atmo_ml/data_utils.py]: when doing this, i got an error in check:
"File "/home/runner/work/climateiq-cnn/climateiq-cnn/usl_models/usl_models/atmo_ml/data_utils.py", line 61, in split_time_step_pairs  *
            B, H, W, T_HALF, C_PAIR = tf.shape(permuted)
    
        OperatorNotAllowedInGraphError: Iterating over a symbolic `tf.Tensor` is not allowed. You can attempt the following resolutions to the problem: If you are running in Graph mode, use Eager execution mode or decorate this function with @tf.function. If you are using AutoGraph, you can try decorating this function with @tf.function. If that does not work, then you may be using an unsupported feature or your source code may not be visible to AutoGraph."

**Pull request merged**